nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/DebugTask.java

```diff
@@ -128,85 +147,114 @@ private List<Configuration> createNewSeedsToRetry() {
         return retryWOtherSeeds;
     }
 
-    private Configuration debugWithConfigurations(List<Configuration> failingConfigurations) {
-        Configuration debConfig = null;
+    private List<Configuration> debugWithConfigurations(List<Configuration> failingConfigurations) {
+        List<Configuration> allDebuggedConfigs = new LinkedList<Configuration>();
         for (Configuration config : failingConfigurations) {
             Configuration dryConfig;
             if ((dryConfig = this.failsOnDry(config)) != null) {
-                Configuration failingConfig = this.startDebugBinary(dryConfig);
-
-                // If debugged down to single choice point, then go ahead and return that
-                if (failingConfig != null && failingConfig.numChoices() == 0) {
-                    return failingConfig;
-                }
-                // Otherwise should go on until finding better one
-                if (debConfig == null || failingConfig.hasFewerChoicePoints(debConfig)) {
-                    debConfig = failingConfig;
-                }
+                // Get all debugged points and just add them to the full list
+                List<Configuration> debuggedConfigs = this.startDebugBinary(dryConfig);
+                allDebuggedConfigs.addAll(debuggedConfigs);
             }
         }
 
-        return debConfig;
+        return allDebuggedConfigs;
     }
 
     private Configuration failsOnDry(Configuration config) {
         return this.failsWithConfig(config, Integer.MIN_VALUE, Integer.MAX_VALUE);
     }
 
-    public Configuration startDebugBinary(Configuration config) {
-        long start = 0;
-        long end = config.getInvocationCount();
+    public List<Configuration> startDebugBinary(Configuration config) {
+        List<Configuration> allFailingConfigurations = new LinkedList<Configuration>();
+
+        List<Pair<Pair<Long, Long>, Configuration>> pairs = new LinkedList<Pair<Pair<Long, Long>, Configuration>>();
+        pairs.add((Pair<Pair<Long, Long>, Configuration>)Pair.of((Pair<Long, Long>)Pair.of(0L,
```

Comment:
Perhaps the initial entry in the list should map the start/end points to the `config`, as it seems the invariant should be that the start/end points map to a failing configuration.